blockchain/message_pool/src/msgpool/selection.rs

```diff
@@ -157,12 +157,16 @@ where
         let mut partitions: Vec<Vec<NodeKey>> = vec![vec![]; MAX_BLOCKS];
         let mut i = 0;
         while i < MAX_BLOCKS && next_chain < chains.len() {
-            let mut gas_limit = fvm_shared::BLOCK_GAS_LIMIT;
+            let mut gas_limit = fvm_shared3::BLOCK_GAS_LIMIT;
             while next_chain < chains.len() {
                 let chain_key = chains.key_vec[next_chain];
                 next_chain += 1;
                 partitions[i].push(chain_key);
-                gas_limit -= chains.get(chain_key).unwrap().gas_limit;
+                let chain_gas_limit = chains.get(chain_key).unwrap().gas_limit;
+                if gas_limit < chain_gas_limit {
+                    break;
+                }
+                gas_limit -= chain_gas_limit;
```

Comment:
Another scary subtraction

Reply:
3 lines above we check that be can do this